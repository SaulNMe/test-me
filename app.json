{
  "expo": {
    "name": "Qualificame",
    "slug": "qualificame-native",
    "privacy": "public",
    "sdkVersion": "33.0.0",
    "platforms": [
      "ios",
      "android"
    ],
    "version": "1.0.0",
    "orientation": "portrait",
    "icon": "./assets/icon.png",
    "splash": {
      "image": "./assets/splashCustom.png",
      "resizeMode": "cover",
      "backgroundColor": "#ffffff",

    },
    "updates": {
      "fallbackToCacheTimeout": 0
    },
    "assetBundlePatterns": [
      "**/*"
    ],
    "ios": {
      "supportsTablet": true
    }
  }
}
