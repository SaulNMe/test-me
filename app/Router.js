import React, { Component } from 'react';
import { Platform, Animated, Easing } from 'react-native';
import { createAppContainer, createStackNavigator, createBottomTabNavigator, createSwitchNavigator } from 'react-navigation';
import WelcomeScreen from 'qualificame-native/app/screens/WelcomeScreen';
import LoginScreen from 'qualificame-native/app/screens/LoginScreen';
import SignUpScreen from 'qualificame-native/app/screens/SignUpScreen';
import CreateAccountScreen from 'qualificame-native/app/screens/CreateAccountScreen';
import SelectPlanScreen from 'qualificame-native/app/screens/SelectPlanScreen';
import PaymentScreen from 'qualificame-native/app/screens/PaymentScreen';
import KiosksScreen from 'qualificame-native/app/screens/KiosksScreen';
import CreateKioskScreen from 'qualificame-native/app/screens/CreateKioskScreen';
import KioskDetailScreen from 'qualificame-native/app/screens/KioskDetailScreen';
import KioskConfigurationScreen from 'qualificame-native/app/screens/KioskConfigurationScreen';
import KioskContactDataScreen from 'qualificame-native/app/screens/KioskContactDataScreen';
import KioskOptionsScreen from 'qualificame-native/app/screens/KioskOptionsScreen';
import ReportsScreen from 'qualificame-native/app/screens/ReportsScreen';
import CreateReportScreen from 'qualificame-native/app/screens/CreateReportScreen';
import SummaryReportScreen from 'qualificame-native/app/screens/SummaryReportScreen';
import ReportDetailScreen from 'qualificame-native/app/screens/ReportDetailScreen';
import MailBoxScreen from 'qualificame-native/app/screens/MailBoxScreen';
import MyAccountScreen from 'qualificame-native/app/screens/MyAccountScreen';
import ProPlanScreen from 'qualificame-native/app/screens/ProPlanScreen';
import EnterprisePlanScreen from 'qualificame-native/app/screens/EnterprisePlanScreen';
import ChangePasswordScreen from 'qualificame-native/app/screens/ChangePasswordScreen';
import FaqScreen from 'qualificame-native/app/screens/FaqScreen';
import AlertsScreen from 'qualificame-native/app/screens/AlertsScreen';
import BottomBar from 'qualificame-native/app/components/BottomBar';
import LoadingScreen from 'qualificame-native/app/screens/LoadingScreen';
import SendEmailScreen from 'qualificame-native/app/screens/SendEmailScreen';
import VerificationCodeScreen from 'qualificame-native/app/screens/VerificationCodeScreen';
import ConfirmPasswordScreen from 'qualificame-native/app/screens/ConfirmPasswordScreen';
import AuthLoadingScreen from 'qualificame-native/app/screens/AuthLoadingScreen';
import ScannerScreen from 'qualificame-native/app/screens/ScannerScreen';

import GetStateForAction from 'qualificame-native/app/utils/GetStateForAction';
import TransitionConfiguration from 'qualificame-native/app/utils/TransitionConfiguration';

const RecoverPasswordStack = createStackNavigator({
	SendEmailScreen,
	VerificationCodeScreen,
	ConfirmPasswordScreen,
	LoadingScreen: { 
		screen: LoadingScreen,
		navigationOptions: null
	},
}, {
	transitionConfig: TransitionConfiguration,
	navigationOptions: {
		header: null
	}
})

const AuthModalStackNavigator = createStackNavigator({
	WelcomeScreen,
	LoginScreen,
	RecoverPasswordStack,
	SignUpScreen,
	CreateAccountScreen,
	SelectPlanScreen,
	PaymentScreen,
	LoadingScreen: { 
		screen: LoadingScreen,
		navigationOptions: null
	},
}, {
	transitionConfig: TransitionConfiguration
});

const KioskConfigurationStack = createStackNavigator({
	KioskConfigurationScreen,
	KioskOptionsScreen,
	KioskContactDataScreen
});

const CreateReportStack = createStackNavigator({
	CreateReportScreen,
	SummaryReportScreen,
	LoadingScreen,
}, {
	transitionConfig: TransitionConfiguration,
	navigationOptions: {
		header: null
	}
});

GetStateForAction(CreateReportStack);

const BottomBarNavigator = createBottomTabNavigator({
	KiosksScreen,
	ReportsScreen,
	MailBoxScreen,
	MyAccountScreen
}, {
	initialRouteName: 'KiosksScreen',
	tabBarComponent: BottomBar
});

const CameraModalNavigator = createStackNavigator({
	ScannerScreen
}, {
	mode: 'modal',
	navigationOptions: {
		header: null,
	}
})

const MainStackNavigator = createStackNavigator({
	BottomBarNavigator: {
		screen: BottomBarNavigator,
		navigationOptions: {
			header: null,
		}
	},
	KioskDetailScreen,
	CameraModalNavigator,
	ReportDetailScreen,
	ProPlanScreen,
	EnterprisePlanScreen,
	ChangePasswordScreen,
	FaqScreen
}, {
	initialRouteName: 'BottomBarNavigator',
});

const ModalStack = createStackNavigator({
	MainStackNavigator,
	AlertsScreen,
	CreateKioskScreen,
	KioskConfigurationStack,
	CreateReportStack
}, {
	mode:'modal',
	headerMode: 'none'
});

const MainNavigation = createSwitchNavigator({
	AuthLoading: AuthLoadingScreen,
	Auth: AuthModalStackNavigator,
	App: ModalStack,
}, {
	initialRouteName: 'AuthLoading'
});

export default createAppContainer(MainNavigation);




 




