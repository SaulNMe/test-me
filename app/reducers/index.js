import Events, * as fromEvents from 'qualificame-native/app/reducers/EventsReducer.js';
import Auth, * as fromAuth from 'qualificame-native/app/reducers/AuthReducer.js';
import TokenRefresh, * as fromTokenRefresh from 'qualificame-native/app/reducers/TokenRefreshReducer.js'
import Kiosks, * as fromKiosks from 'qualificame-native/app/reducers/KiosksReducer.js';
import Reports, * as fromReports from 'qualificame-native/app/reducers/ReportsReducer.js';
import Choices, * as fromChoices from 'qualificame-native/app/reducers/ChoicesReducer.js';
import Alerts, * as fromAlerts from 'qualificame-native/app/reducers/AlertsReducer.js';

import { combineReducers } from 'redux'

const rootReducer = combineReducers({
    Auth,
    TokenRefresh,
    Kiosks,
    Reports,
    Events,
    Choices,
    Alerts,
});

export default rootReducer;

//Auth
export const getIsLoggedIn = state =>
	fromAuth.getIsLoggedIn(state.Auth);

export const getIsLoading = state =>
	fromAuth.getIsLoading(state.Auth);

export const getAuthData = state =>
	fromAuth.getAuthData(state.Auth);

export const getUserData = state =>
	fromAuth.getUserData(state.Auth);

//Token Refresh
export const getIsRefreshingToken = state =>
	fromTokenRefresh.getIsRefreshingToken(state.TokenRefresh);

export const getRefreshCall = state =>
	fromTokenRefresh.getRefreshCall(state.TokenRefresh);

//Kiosks
export const getKiosks = state =>
	fromKiosks.getKiosks(state.Kiosks);

export const getKioskById = (state, kioskId) =>
	fromKiosks.getKioskById(state.Kiosks, kioskId)

export const getIsLoadingKiosks = state =>
	fromKiosks.getIsLoadingKiosks(state.Kiosks);

export const getErrorKiosks = state =>
	fromKiosks.getErrorKiosks(state.Kiosks);

//Reports

export const getReports = state =>
	fromReports.getReports(state.Reports);

export const getIsLoadingReports = state =>
	fromReports.getIsLoadingReports(state.Reports);
export const getErrorReports = state =>
	fromReports.getErrorReports(state.Reports);

// Events
export const getEvents = state =>
	fromEvents.getEvents(state.Events);

export const getEventsGroupedByDay = (state, ids) =>
	fromEvents.getEventsGroupedByDay(state.Events, ids);

export const getSelectedMonth = state => 
	fromEvents.getSelectedMonth(state.Events);

export const getEventsSelectedDateRange = state => 
	fromEvents.getEventsSelectedDateRange(state.Events);

// Choices
export const getChoices = state =>
	fromChoices.getChoiceById(state.Choices);

// Alerts 
export const getAlerts = state =>
	fromAlerts.getAlerts(state.Alerts);

export const unReadAlerts = state =>
	fromAlerts.unReadAlerts(state.Alerts);