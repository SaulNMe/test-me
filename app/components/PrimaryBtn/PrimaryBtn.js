import React, { PureComponent } from 'react';
import { Text, View, TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';
import BodyText from 'qualificame-native/app/components/BodyText';
import { LinearGradient } from 'expo-linear-gradient';
import { Colors } from 'qualificame-native/app/styles';


import styles from './PrimaryBtnStyle';

export default class PrimaryBtn extends PureComponent {
	render() {
		let textColor;
		let gradientColors;
		let disabledGradient;
		switch(this.props.bgColor) {
			case 'green':
				gradientColors = ['#0F9F92', '#12B184'];
				disabledGradient = ['#58c2ae', '#a0ddd0'];
				textColor = "white";
			break;
			case 'blue':
				gradientColors = ['#154BBA', '#107EC2'];
				disabledGradient = ['#5792d1', '#a0c3e5'];
				textColor = "white";
			break;
			case 'orange':
				gradientColors = ['#E62808', '#FC680A'];
				disabledGradient = ['#f58053', '#fab69d'];
				textColor = "white";
			break;
			case 'pink':
				gradientColors = ['#E31862', '#EB194D'];
				disabledGradient = ['#ec5c88', '#f3a0b8'];
				textColor = "white";
			break;
			default:
				gradientColors = ['#fff', '#fff'];
				disabledGradient = ['#CCCACA', '#E6E3E3'];
				textColor = "dark";
			break;
		}
		return (
			<TouchableOpacity
				onPress={this.props.onPress}
				disabled={this.props.disabled}
				activeOpacity={0.9}
			>
				<View style={[styles.borderRadius, (this.props.hasHorizontalMargin && styles.baseHorizontalMargin), styles.centerObjects, (!this.props.disabled && styles.shadow)]}>
					<LinearGradient colors={this.props.disabled ? disabledGradient : gradientColors} style={[styles.fullWidth, styles.defaultStyles, styles.alignItemsCenter]} start={[0, 0]} end={[1, 0]}>
						<BodyText 
							text={this.props.text}
							color={textColor}
							weight={this.props.weight}
						/>
					</LinearGradient>
				</View>
			</TouchableOpacity>
		);
	}
}

	PrimaryBtn.propTypes = {
		onPress: PropTypes.func,
		text: PropTypes.string,
		bgColor: PropTypes.string,
		disabled: PropTypes.bool,
		hasHorizontalMargin: PropTypes.bool,
		weight: PropTypes.string,
	}

	PrimaryBtn.defaultProps = {
		onPress: ()=>{},
		text: "Click me!",
		bgColor: "",
		disabled: false,
		hasHorizontalMargin: false,
		weight: 'medium'
	}


