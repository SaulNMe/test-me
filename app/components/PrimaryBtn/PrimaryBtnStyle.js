import { StyleSheet } from 'react-native';
import { ApplicationStyles, Fonts, Colors, Metrics } from 'qualificame-native/app/styles';

export default StyleSheet.create({
	...ApplicationStyles.flexBox,
	...ApplicationStyles.screen,
	...ApplicationStyles.margins,
	defaultStyles: {
		borderRadius: 10,
		padding: Metrics.smallMargin,
	},
	shadow: {
		elevation: 2,
		shadowColor: '#000',
		shadowRadius: 4,
		shadowOpacity: 0.5,
		shadowOffset: { width: 0, height: 4 }
	},
	disabledText: {
		color: Colors.dark
	},
	disabledBg: {
		backgroundColor: Colors.disabled
	},
	borderRadius: {
		borderRadius: 10
	}
});
