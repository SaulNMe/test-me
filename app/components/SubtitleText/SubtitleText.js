import React, { PureComponent } from 'react';
import { Text } from 'react-native';
import PropTypes from 'prop-types';
import { Fonts, Colors } from 'qualificame-native/app/styles';
import styles from './SubtitleTextStyle';

export default class SubtitleText extends PureComponent {
	render() {
		const textColor = { color: Colors[this.props.color] };
		const weight = { fontWeight: Fonts.weight[this.props.weight] };
		const align = { textAlign: this.props.align };
		return (
			<React.Fragment>
				<Text style={[styles.text, textColor, weight, align]}>{this.props.text}</Text>
			</React.Fragment>
		);
	}
}

	SubtitleText.propTypes = {
		text: PropTypes.string,
		color: PropTypes.string,
		weight: PropTypes.string,
		align: PropTypes.string,
	}

	SubtitleText.defaultProps = {
		text: "",
		color: "dark",
		weight: "regular",
		align: "left"
	}
