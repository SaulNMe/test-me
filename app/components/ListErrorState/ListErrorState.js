import React, { PureComponent } from 'react';
import { Text, View } from 'react-native';
import PropTypes from 'prop-types';

import styles from './ListErrorStateStyle';
import SubtitleText from 'qualificame-native/app/components/SubtitleText';
import PrimaryBtn from 'qualificame-native/app/components/PrimaryBtn';

export default class ListErrorState extends PureComponent {
	render() {
		return (
			<View style={[styles.fullWidth, styles.baseHorizontalPadding, styles.baseVerticalMargin]}>
				<View style={[styles.row, styles.justifyContentCenter]}>
					<SubtitleText
						align='center'
						text={this.props.text}
						color='dark'
					/>
				</View>
{/*				<View style={[styles.baseVerticalMargin]}>
					<PrimaryBtn 
						onPress={this.props.onPress}
						text={this.props.btnText}
						bgColor={this.props.color}
						hasHorizontalMargin
					/>
				</View>*/}
			</View>
		);
	}
}

	ListErrorState.propTypes = {
		onPress: PropTypes.func,
		color: PropTypes.string,
		text: PropTypes.string,
		btnText: PropTypes.string,
	}

	ListErrorState.defaultProps = {
		onPress: ()=>{},
		color: '',
		text: "Parece que ocurrió un error. Vuelve a intentar más tarde",
		btnText: '',
	}
