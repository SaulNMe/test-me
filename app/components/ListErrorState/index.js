import ListErrorState from './ListErrorState';

export default ListErrorState;
