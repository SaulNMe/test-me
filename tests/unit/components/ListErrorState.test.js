import React from 'react';
import { shallow } from 'enzyme';

import ListErrorState from 'qualificame-native/app/components/ListErrorState';
import SubtitleText from 'qualificame-native/app/components/SubtitleText';
import PrimaryBtn from 'qualificame-native/app/components/PrimaryBtn';

import { Text } from 'react-native';
import { Colors } from 'qualificame-native/app/styles';


function createTestProps (props) {
	return {
		// common props
		color: 'blue',
		text: 'Ups... Parece que hubo un error',
		btnText: 'Volver a intentar',
		// allow to override common props
		...props,
	}
}

describe('rendering', () => {
	let wrapper;
	const createWrapper = props => shallow(<ListErrorState {...props}/>);
	beforeEach(() => {
		const props = createTestProps()
		wrapper = createWrapper(props);
	});

	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('displays the correct text ', () => {
		expect(wrapper.find(SubtitleText).prop('text')).toEqual('Ups... Parece que hubo un error');
	});	

	it('displays the correct PrimaryBtn bgColor ', () => {
		expect(wrapper.find(PrimaryBtn).prop('bgColor')).toEqual('blue');
	});

	it('displays a PrimaryBtn', () => {
		expect(wrapper.find(PrimaryBtn).exists()).toBeTruthy();
	});	

	it('displays the correct PrimaryBtn text', () => {
		expect(wrapper.find(PrimaryBtn).exists()).toBeTruthy();
		expect(wrapper.find(PrimaryBtn).prop('text')).toEqual('Volver a intentar');
	});

});