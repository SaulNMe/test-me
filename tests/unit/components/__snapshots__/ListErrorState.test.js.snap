// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders as expected 1`] = `
<Component
  style={
    Array [
      Object {
        "width": "100%",
      },
      Object {
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <Component
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "justifyContent": "center",
        },
      ]
    }
  >
    <SubtitleText
      align="center"
      color="dark"
      text="Ups... Parece que hubo un error"
      weight=""
    />
  </Component>
  <Component
    style={
      Array [
        Object {
          "marginVertical": 16,
        },
      ]
    }
  >
    <PrimaryBtn
      bgColor="blue"
      disabled={false}
      hasHorizontalMargin={true}
      onPress={[Function]}
      text="Volver a intentar"
    />
  </Component>
</Component>
`;
